smarts/core/smarts.py

```diff
@@ -1354,10 +1353,15 @@ def _try_emit_envision_state(self, provider_state: ProviderState, obs, scores):
                         for paths in vehicle_obs.road_waypoints.lanes.values()
                         for path in paths
                     ]
+
+                veh_type = (
+                    v.vehicle_config_type if v.vehicle_config_type else v.vehicle_type
+                )
+                
                 traffic[v.vehicle_id] = envision_types.TrafficActorState(
                     name=self._agent_manager.agent_name(agent_id),
                     actor_type=actor_type,
-                    vehicle_type=envision_types.VehicleType.Car,
+                    vehicle_type=veh_type,
```

Comment:
This change seems to cause the ego vehicle to be the wrong color (grey instead of red) on SUMO maps